tests/api-watch/test_diff_class.ml

```diff
@@ -0,0 +1,76 @@
+open Api_watch
+open Test_helpers
+
+let%expect_test "Class Addition" =
+  let reference =
+    {|
+  class cls1 : object
+    method m1 : int -> int
+  end
+  class cls2 : object
+    method m2 : string -> string
+  end
+|}
+  in
+  let current =
+    {|
+  class cls1 : object
+    method m1 : int -> int
+  end
+  class cls2 : object
+    method m2 : string -> string
+  end
+  class cls3 : object
+    method m3 : float -> float
+  end
+|}
+  in
+  try
+    let ref_compiled = compile_interface reference in
+    let curr_compiled = compile_interface current in
+    let result =
+      Diff.interface ~module_name:"Main" ~reference:ref_compiled
+        ~current:curr_compiled
+    in
+    Format.printf "%a" pp_diff_option result;
+    [%expect
+      {|
+    Some (Module Main: {Modified (Supported [ Type (cls3, Added);
+    Class (cls3, Added)])})
```

Comment:
As you can see here, there is a type added in addition to the class addition.
This is because of the way classes are added in a signature in OCaml, so that is normal, but it will be unexpected by most OCaml users.

(There is nothing to act on in this comment. I'll open a new issue to fix this when this PR is merged).